src/core/utilities/generate_aliases.ts

```diff
@@ -1,3 +1,133 @@
+function cartesianProduct<T>(arr: T[][]): T[][] {
+  return arr.reduce(
+    (a, b) => {
+      return a
+        .map(x => {
+          return b.map(y => {
+            return x.concat(y);
+          });
+        })
+        .reduce((c, d) => c.concat(d), []);
+    },
+    [[]] as T[][]
+  );
+}
+
+export function* aliasesFromPattern(query: string) {
+  if (!query || query.length === 0) {
+    throw new Error('Tried to parse null or empty string');
+  }
+
+  let i = 0;
+  const currentChar = () => query[i];
+
+  function match(c: string) {
+    const char = currentChar();
+    if (c.includes(char)) {
+      i++;
+      return char;
+    }
+    else if (i >= query.length) {
+      throw new Error(
+        `Ran off the end of the string expecting to find: '${c}', i: ${i}`
+      );
+    } else {
+      throw new Error(
+        `Unexpected character: '${currentChar()}'. Expected '${c}'.`
+      );
+    }
+  }
+
+  //Intrinsics
+  const whitespace = () => match(' ');
+  const letter = () =>
+    match(
```

Comment:
Consider using on of the built-in iswhitespace methods here to handle other locals.